client/javascript/scrabble.js

```diff
@@ -662,17 +976,28 @@ function calculateMove(squares)
     legalPlacements[topLeftX][topLeftY] = true;
 
     function touchingOld(x, y) {
+	    switch(boardLen) {
+		    case 21:
         var retval = 
         (x > 0 && squares[x - 1][y].tile && squares[x - 1][y].tileLocked)
-            || (x < 14 && squares[x + 1][y].tile && squares[x + 1][y].tileLocked)
+            || (x < 20 && squares[x + 1][y].tile && squares[x + 1][y].tileLocked)
             || (y > 0 && squares[x][y - 1].tile && squares[x][y - 1].tileLocked)
+            || (y < 20 && squares[x][y + 1].tile && squares[x][y + 1].tileLocked);
+			    break;
+		    default:
+        var retval = 
+        (x > 0 && squares[x - 1][y].tile && squares[x - 1][y].tileLocked)
+            || (x < 14 && squares[x + 1][y].tile && squares[x + 1][y].tileLocked)
+             || (y > 0 && squares[x][y - 1].tile && squares[x][y - 1].tileLocked)
             || (y < 14 && squares[x][y + 1].tile && squares[x][y + 1].tileLocked);
+			    break;
+	    }
```

Comment:
This looks to me as if this could be much simpler - instead of 14 or 20, `boardLen - 1` could be used and the case would then not be needed.

Reply:
simplified.